cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/callables/CreateIndexCallable.java

```diff
@@ -0,0 +1,144 @@
+            statements.add(createVirtualTableStatementForIndex(index.indexName,
+                    columnList,
+                    Collections.singletonList(settings)));
+        } else {
+            statements.add(createIndexTableStatementForIndex(index.indexName, columnList));
+            statements.add(createIndexIndexStatementForIndex(index.indexName, columnList));
+        }
+        for (String statement : statements) {
+            try {
+                database.execSQL(statement);
+            } catch (SQLException e) {
+                String msg = String.format("Index creation error occurred (%s):", statement);
+                throw new QueryException(msg, e);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * This method generates the virtual table create SQL for the specified index.
+     * Note:  Any column that contains an '=' will cause the statement to fail
+     * because it triggers SQLite to expect that a parameter/value is being passed in.
+     *
+     * @param indexName     the index name to be used when creating the SQLite virtual table
+     * @param columns       the columns in the table
+     * @param indexSettings the special settings to apply to the virtual table -
+     *                      (only 'tokenize' is current supported)
+     * @return the SQL to create the SQLite virtual table
+     */
+    private String createVirtualTableStatementForIndex(String indexName,
+                                                       List<String> columns,
+                                                       List<String> indexSettings) {
+        String tableName = String.format(Locale.ENGLISH, "\"%s\"", QueryImpl
+                .tableNameForIndex(indexName));
+        String cols = Misc.join(",", columns);
+        String settings = Misc.join(",", indexSettings);
+
+        return String.format("CREATE VIRTUAL TABLE %s USING FTS4 ( %s, %s )", tableName,
+                cols,
+                settings);
+    }
+
+    private String createIndexTableStatementForIndex(String indexName, List<String> columns) {
+        String tableName = String.format(Locale.ENGLISH, "\"%s\"", QueryImpl.tableNameForIndex(indexName));
+        String cols = Misc.join(" NONE, ", columns);
+
+        return String.format("CREATE TABLE %s ( %s NONE )", tableName, cols);
+    }
+
+    private String createIndexIndexStatementForIndex(String indexName, List<String> columns) {
+        String tableName = QueryImpl.tableNameForIndex(indexName);
+        String sqlIndexName = tableName.concat("_index");
+        String cols = Misc.join(",", columns);
+
+        return String.format(Locale.ENGLISH, "CREATE INDEX \"%s\" ON \"%s\" ( %s )", sqlIndexName, tableName, cols);
+    }
+
+    /**
+     * Helper to convert the index settings to a JSON string
+     *
+     * @return the JSON representation of the index settings
+     */
+    public static String settingsAsJSON(Index i) {
```

Comment:
It can be, the only external caller is the test code, I didn't want to move it initially to keep the size down, but I'll go ahead and do that.